src/toxcore/dht/server/mod.rs

```diff
@@ -954,9 +943,9 @@ impl Server {
     fn handle_dht_req_for_others(&self, packet: DhtRequest)
         -> impl Future<Item = (), Error = HandlePacketError> + Send {
         let close_nodes = self.close_nodes.read();
-        if let Some(node) = close_nodes.get_node(&packet.rpk) { // search close_nodes to find target peer
+        if let Some(node) = close_nodes.get_node(&packet.rpk).and_then(|node| node.to_packed_node()) {
```

Comment:
Better add one more combinator instead of `if let Some`:

```
.then(|maybe_node| {
   if let Some(node) = maybe_node {
   } else {
       Either::B(future::ok(()))

   }
}
```

Reply:
Your code won't work - it has to be inner `if let` in this case which I don't like. I can move it to separate variable if you'd like.

Reply:
> Your code won't work - it has to be inner `if let` in this case which I don't like.

Either you **don't** like it or code **won't** work. Please choose one option.

Reply:
It won't work just because `Option` doesn't have `then` method. So I had to make an assumption what did you mean an I supposed that it's two inner `if let`, but I can be wrong.

Reply:
> just because `Option` doesn't have `then` method.

OK